server/domLoader.js

```diff
@@ -3,13 +3,13 @@ const puppeteer = require('puppeteer');
 
 module.exports = class DomLoader {
 
-  static async getHTMLfromURL(url) {
-    let result = await DomLoader.runBrowser(url);
-    if (result.error) result = await DomLoader.runBrowser(url);
+  static async getHTMLfromURL(url, selector = null) {
+    let result = await DomLoader.runBrowser(url, selector);
+    if (result.error) result = await DomLoader.runBrowser(url, selector);
     return result;
   }
 
-  static async runBrowser(url) {
+  static async runBrowser(url, selector) {
```

Comment:
`selector` is used here to specify the concrete html DOM element you want to parse?

Reply:
yes